Feature/sdl 0264 separating the change of audible status and the change of hmi status (#251)
smartdevicelink/sdl_hmi_integration_guidelines: docs/BasicCommunication/OnEventChanged/index.md

!!! MUST
1. Send notification with appropriate parameter value when active call on HMI has been started/ended.

> 2. Resume the applications to their original audio streaming state prior to the phone call event in the HMI when the event ends (see note below).

Reviewer: ```suggestion
2. Return the applications to their appropriate audio streaming state in the HMI when the event ends.
```
Realizing this is slightly misleading, since a different app could have been activated during the event, meaning that we shouldn't resume the state if it's no longer applicable.

!!!

!!! NOTE
If the HMI wants to leave the app screen when the PHONE_CALL event occurs, the app's HMI state can be changed accordingly using the `BC.OnAppDeactivated(appID)` and `BC.OnAppActivated(appID)` notifications.
!!!

Upon receiving `OnEventChanged(PHONE_CALL)`, SDL will:

|isActive|Result|
|:-------|:-----|

> |true|Keep the HMI state of all applications but change audible state of all applications to NOT_AUDIBLE|

Reviewer: ```suggestion
|true|Move all apps with audioStreamingState AUDIBLE to NOT_AUDIBLE|
```
I still think this should be modified to be consistent with the rest of the document (see `EMERGENCY_EVENT` section). I don't think that the wording in the proposal needs to be followed strictly in this case, it seems to be more of an example.

Author: @jacobkeeler -san,
Thank you for your comment.
We understood.
We made a commit your suggestion.
Thank you.

> |false|Return applications to the same HMI state they had prior to the event|

Reviewer: ```suggestion
|false|Return applications to the same HMI state they had prior to the event (if not changed during the event)|
```

Author: @jacobkeeler -san,
Basically, we will accept your suggestion.
However, we have two question.
1.
Our proposal is to apply to only case of "isActive = true".
Therefore, we assume that the case of "is Active = false" is not applicable.
2.
We left [this comment](https://github.com/smartdevicelink/sdl_hmi/pull/576#issuecomment-880328403).
If need to modify this, we think that it is necessary to add the case of changing the last application during the event.

For these reasons, we ccouldnot make a commit it.
If you have some idea, could you please teach us.
Thank you.

Reviewer: 1. Hmmm, thinking on it, this is actually more of an existing issue with the documentation, not necessarily tied to this proposal. Perhaps I'll create a separate PR for this, since it would probably affect the other states as well.
